packages-ts/starknet/src/network.ts

```diff
@@ -0,0 +1,159 @@
+import { ChildProcess, spawn } from 'child_process'
+import net from 'net'
+import { DEVNET_NAME, HARDHAT_NAME } from './utils'
+
+export class NetworkManager {
+  private opts?: any
+  private strategyDevnet: StarknetDevnet
+  private strategyHardhat: HardHatNode
+
+  constructor(opts?: any) {
+    this.opts = opts
+    if (this.opts?.required[0] === 'starknet') {
+      if (this.opts.config.starknet === DEVNET_NAME) {
+        this.strategyDevnet = new StarknetDevnet('5050', this.opts)
+      }
+    }
+    if (this.opts?.required[1] === 'ethereum') {
+      if (this.opts.config.ethereum === HARDHAT_NAME) {
+        this.strategyHardhat = new HardHatNode('8545', this.opts)
+      }
+    }
```

Comment:
This implementation is prone to bugs...

We should make sure that all required networks are managed, or throw an error.
We should make sure that a network is declared only once, or throw an error.
We don't care about the order at which they are declared.